File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> count;
        for (const int n: nums) {
            count[n]++;
            if (count[n] > nums.size() / 2) {
                return n;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        d = defaultdict(int)
        l = len(nums)
        res = set()

        for n in nums:
            d[n] += 1
            if d[n] > l / 3:
                res.add(n)
        return list(res) 
